memory-db/src/lib.rs

```diff
@@ -698,4 +730,18 @@ mod tests {
 		assert!(db2.contains(&root, EMPTY_PREFIX));
 		assert!(db.contains(&root, EMPTY_PREFIX));
 	}
+
+	#[test]
+	fn malloc_size_of() {
+		let mut db = MemoryDB::<KeccakHasher, HashKey<_>, Vec<u8>>::default();
+		for i in 0u32..1024 {
+			let bytes = i.to_be_bytes();
+			let prefix = (bytes.as_ref(), None);
+			db.insert(prefix, &bytes);
+		}
+		assert_eq!(
+			malloc_size(&db),
+			malloc_size(&db.data) + malloc_size(&db.null_node_data) + malloc_size(&db.hashed_null_node)
+		);
+	}
```

Comment:
this would better be a property test via `quickcheck` or a fuzzer, but I was hesitant to introduce a new dependency just for one test